kiwi/package_manager/base.py

```diff
@@ -182,3 +183,8 @@ def cleanup_requests(self):
         del self.package_requests[:]
         del self.collection_requests[:]
         del self.product_requests[:]
+
+    def __del__(self):
+        log.info('Cleaning up all repos from %s instance', type(self).__name__)
+        if self.repository is not None:
```

Comment:
I would suggest:
```python
if self.repository:
    self.repository.delete_all_repos()
```
Because `self.repository` supposed to be an object, specifig to repos. Although this is still pretty odd, because normally it would be:
```python
self.repository.delete()
```
...or:
```python
for repo in self.repositories:
    repo.delete()
```